readthedocs/core/adapters.py

```diff
@@ -50,3 +55,16 @@ def save_user(self, request, user, form, commit=True):
                 invitation.delete()
             else:
                 log.info("Invitation not found", invitation_pk=invitation_pk)
+
+
+class SocialAccountAdapter(DefaultSocialAccountAdapter):
+
+    def pre_social_login(self, request, sociallogin):
+        provider = sociallogin.account.get_provider()
+        if provider.id == GitHubAppProvider.id and not sociallogin.is_existing:
```

Comment:
What `is_existing` does? I didn't find the documentation. Does this mean whether or not the user has already a GitHubApp created?

Reply:
Re-reading the code a little more deeper, I understand: "if the user is logging in with GitHub and doesn't have GitHub App already connected to their account".

In that case, we get the user from the old GitHub OAuth account and automatically connect the new GitHub App to their account.

That sounds 👍🏼  to me.